packages/client/src/generation/generateClient.ts

```diff
@@ -539,5 +545,18 @@ async function copyRuntimeFiles({ from, to, runtimeName, sourceMaps }: CopyRunti
     files.push(...files.filter((file) => file.endsWith('.js')).map((file) => `${file}.map`))
   }
 
-  await Promise.all(files.map((file) => fs.promises.copyFile(path.join(from, file), path.join(to, file))))
+  await Promise.all(files.map((file) => fs.copyFile(path.join(from, file), path.join(to, file))))
+}
+
+/**
+ * Attempts to delete the output directory.
+ * @param finalOutputDir
+ */
+async function deleteOutputDir(finalOutputDir: string) {
+  try {
+    // we want to make sure that if we delete, we delete the right directory
+    if (require(`${finalOutputDir}/package.json`).name === GENERATED_PACKAGE_NAME) {
+      await fs.rmdir(finalOutputDir, { recursive: true })
+    }
+  } catch {}
```

Comment:
If we silence this and that somehow the deletion doesn't succeed, it would look like `--no-engine` did nothing? 
It would be hard to debug a case like this?

Reply:
At least some debug log that surfaces his would be good.

Reply:
Added internal debug logs